lib/util/git/index.ts

```diff
@@ -788,8 +791,13 @@ export async function mergeBranch(branchName: string): Promise<void> {
       ])
     );
     status = await git.status();
-    await gitRetry(() => git.merge(['--ff-only', branchName]));
-    await gitRetry(() => git.push('origin', config.currentBranch));
+    if (localOnly) {
+      // merge commit, don't push to origin
+      await gitRetry(() => git.merge([branchName]));
+    } else {
+      await gitRetry(() => git.merge(['--ff-only', branchName]));
+      await gitRetry(() => git.push('origin', config.currentBranch));
```

Comment:
```suggestion
    await gitRetry(() => git.merge(['--ff-only', branchName]));
    if (!localOnly) {
      await gitRetry(() => git.push('origin', config.currentBranch));
```
Why do we create a merge commit if running in local mode?

Reply:
Because the onboarding branch may be behind base branch and possibly modified too. Ff in such cases may not be possible.